doc/rvv-intrinsic-spec.adoc

```diff
@@ -1,165 +1,180 @@
+The RISC-V vector intrinsics data types are strongly-typed, and hence so are the intrinsics taking these types as parameters. The vector intrinsics encode the EEW (effective-element-width) ^17^ and EMUL (effective LMUL) ^17^ of the destination vector register into the suffix of the function name. Users can expect the results of the vector instruction intrinsics are computed under the specified EEW and EMUL.
 
-To see the full list of data types for the intrinsics, please see <<chapter-type-system>>.
+To see the full list of data types for the intrinsics, please see <<type-system>>.
 
-In the following example, the intrinsic will produce the result of `vadd.vv` with vector operands of `EEW=32` and `EMUL=1`.
+In the following example, the intrinsic will produce the semantic of a `vadd.vv` instruction, with source vector operands of `EEW=32` and `EMUL=1`, which can be observed through provided RVV data type; and produce results to destination vector operands with `EEW=32` and `EMUL=1`.
 
-```
+[,c]
+----
 vint32m1_t __riscv_vadd_vv_i32m1 (vint32m1_t op1, vint32m1_t op2, size_t vl);
-```
+----
 
-In the following example, the intrinsic will produce the result of `vwadd.vv` with destinations of `EEW=32` and `EMUL=1`, and operands of `EEW=16` and `EMUL=mf2 (1/2)`.
+In the following example, the intrinsic will produce the semantic of a `vadd.vv` instruction, with source vector operands of `EEW=16` and `EMUL=1/2`, which can be observed through provided RVV data type; and produce results to destination vector operands with `EEW=32` and `EMUL=1`.
 
-```
+[,c]
+----
 vint32m1_t __riscv_vwadd_vv_i32m1 (vint16mf2_t op1, vint16mf2_t op2, size_t vl);
-```
+----
 
+[[control-of-vl]]
 === Control of `vl`
 
-Direct control of `vl` (vector length register) ^11^  is not exposed at the C language level. In the intrinsics that represents vector instructions, the `vl` operand indicates the active vector length valid during intrinsic execution.
+Direct control of `vl` (vector length register) ^11^  is not exposed at the C language level. The RVV C intrinsics provide intrinsics with a `vl` operand to indicate the active vector length valid during intrinsic execution. The compiler is responsible to set the correct value to `vl` given the informed active vector length.
 
 NOTE: Intrinsics for instructions that behave the same with different `vl` settings (e.g. `vmv.s.x`) do not have a `vl` operand.
 
-NOTE: Intrinsics will at most operate `vlmax` (derived in 3.4.2 of v-spec) elements.
+NOTE: The intrinsics follow the specification ^27^ and will at most operate on `VLMAX` ^5^ elements. If a value larger than `VLMAX` is provided to `vl`, the intrinsics will operate on `VLMAX` elements. Essentially the number of elements processed will match what is returned when provided to `__riscv_vsetvl_*` <<pseudo-vsetvl>>.
```

Comment:
This is not exactly correct, as discussed by @dzaima and @topperc in the PR conversation. In this case, I think it is better to say less. Here is a proposed fix.

```suggestion
The actual value written to the `vl` control register is subject to implementation defined behavior and is typically not known until runtime.
The intrinsics programmer should use the `__riscv_vsetvl_*` intrinsics <<pseudo-vsetvl>> to determine its value,
for use (e.g.) in loop stripmining logic.
```

Reply:
On second thought, this might warrant further discussion. For known VLEN or sufficiently small AVL, the programmer might want to avoid stripmining and rely on the fact that `vl` = AVL when AVL < VLMAX. The way I've worded this might allow an implementation to pick `vl` < AVL in this case.

Reply:
I just made an edit to my proposed fix. I'm writing this message so you see a notification about it.